models/classes/Lists/DataAccess/Repository/DependsOnPropertyRepository.php

```diff
@@ -61,33 +71,64 @@ public function findAll(array $options): DependsOnPropertyCollection
+        if (is_null($dependentsOnPropertyJson)) {
+            $dependencies = $this->getDependencies($listUri);
+            if (empty($dependencies)) {
+                return $collection;
             }
 
-            if (!$this->getDependentPropertySpecification()->isSatisfiedBy($classProperty)) {
-                $collection->append(new DependsOnProperty($classProperty));
+            $dependencyListUris = $this->getDependencyListUris($dependencies);
+            if (empty($dependencyListUris)) {
+                return $collection;
+            }
 
-                continue;
+            $parentPropertiesList = $this->getpropertiesByListUris($dependencyListUris);
+            if (empty($parentPropertiesList)) {
+                return $collection;
             }
 
-            // @TODO Check for parent's (current property) children outside the foreach statement
-            if ($propertyUri === $classProperty->getDependsOnPropertyCollection()->current()->getUri()) {
-                return new DependsOnPropertyCollection();
+            /** @var core_kernel_classes_Class $class */
+            $class = $property->getDomain()->get(0);
+            $propertyUri = $property->getUri();
+
+            /** @var core_kernel_classes_Property $property */
+            foreach ($this->getProperties($class) as $classProperty) {
+                if (
+                    $propertyUri === $classProperty->getUri()
+                    || !$this->getRemoteListPropertySpecification()->isSatisfiedBy($classProperty)
+                ) {
+                    continue;
+                }
+                if (
+                    !$this->getDependentPropertySpecification()->isSatisfiedBy($classProperty)
+                    && in_array($classProperty->getUri(), $parentPropertiesList, true)
+                ) {
+                    $collection->append(new DependsOnProperty($classProperty));
+
+                    continue;
+                }
+
+                // @TODO Check for parent's (current property) children outside the foreach statement
+                // if ($propertyUri === $classProperty->getDependsOnPropertyCollection()->current()->getUri()) {
+                //     return new DependsOnPropertyCollection();
+                // }
             }
+            $this->getCache()->set(self::CACHE_PREFIX . $property->getUri() . "_" . $listUri, json_encode($collection));
+            return $collection;
+        }
+        if (!empty($dependentsOnPropertyJson)) {
+            $dependentsOnPropertyArray = json_decode($dependentsOnPropertyJson);
+            if (!empty($dependentsOnPropertyArray)) {
+                foreach ($dependentsOnPropertyArray as $element) {
+                    $property = new core_kernel_classes_Property($element->uri);
+                    $collection->append(new DependsOnProperty($property->getProperty($element->uri)));
+                }
+            }
+            return $collection;
```

Comment:
Lets try to make the code smaller :) 

```suggestion

    $dependentsOnPropertyArray = json_decode($dependentsOnPropertyJson);

    foreach ($dependentsOnPropertyArray as $element) {
        $property = new core_kernel_classes_Property($element->uri);
        $collection->append(new DependsOnProperty($property->getProperty($element->uri)));
     }
            
     return $collection;
```